modules/dag-l1/src/test/scala/org/tessellation/dag/l1/domain/block/BlockServiceSuite.scala

```diff
@@ -0,0 +1,140 @@
+import org.tessellation.dag.domain.block.generators._
+import org.tessellation.dag.l1.Main
+import org.tessellation.dag.l1.domain.address.storage.AddressStorage
+import org.tessellation.dag.l1.domain.block.BlockStorage
+import org.tessellation.dag.l1.domain.block.BlockStorage._
+import org.tessellation.dag.l1.domain.transaction.TransactionStorage
+import org.tessellation.dag.l1.domain.transaction.TransactionStorage.Majority
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.schema._
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.balance.{Amount, Balance}
+import org.tessellation.schema.transaction._
+import org.tessellation.sdk.sdkKryoRegistrar
+import org.tessellation.security.Hashed
+import org.tessellation.security.hash.ProofsHash
+import org.tessellation.security.signature.Signed
+
+import eu.timepit.refined.auto._
+import io.chrisdavenport.mapref.MapRef
+import weaver.MutableIOSuite
+import weaver.scalacheck.Checkers
+
+object BlockServiceSuite extends MutableIOSuite with Checkers {
+
+  type Res = KryoSerializer[IO]
+
+  override def sharedResource: Resource[IO, BlockServiceSuite.Res] =
+    KryoSerializer.forAsync[IO](Main.kryoRegistrar ++ sdkKryoRegistrar)
+
+  def mkBlockService(
+    blocksR: MapRef[IO, ProofsHash, Option[StoredBlock]],
+    lastAccTxR: MapRef[IO, Address, Option[TransactionStorage.LastTransactionReferenceState]],
+    notFoundParentRef: Option[BlockReference] = None
+  )(implicit K: KryoSerializer[IO]) = {
+
+    val blockAcceptanceManager = new BlockAcceptanceManager[IO]() {
+
+      override def acceptBlocksIteratively(blocks: List[Signed[DAGBlock]], context: BlockAcceptanceContext[IO]): IO[BlockAcceptanceResult] =
+        ???
+
+      override def acceptBlock(
+        block: Signed[DAGBlock],
+        context: BlockAcceptanceContext[IO]
+      ): IO[Either[BlockNotAcceptedReason, (BlockAcceptanceContextUpdate, UsageCount)]] = IO.pure(notFoundParentRef).map {
+        case None      => (BlockAcceptanceContextUpdate.empty, initUsageCount).asRight[BlockNotAcceptedReason]
+        case Some(ref) => ParentNotFound(ref).asLeft[(BlockAcceptanceContextUpdate, UsageCount)]
+      }
+    }
+
+    val addressStorage = new AddressStorage[IO] {
+
+      override def getBalance(address: Address): IO[balance.Balance] = ???
+
+      override def updateBalances(addressBalances: Map[Address, Balance]): IO[Unit] = IO.unit
+
+      override def clean: IO[Unit] = ???
+
+    }
+    Random.scalaUtilRandom.flatMap { implicit r =>
+      MapRef.ofConcurrentHashMap[IO, Address, NonEmptySet[Hashed[Transaction]]]().map { waitingTxsR =>
+        val blockStorage = new BlockStorage[IO](blocksR)
+        val transactionStorage = new TransactionStorage[IO](lastAccTxR, waitingTxsR)
+        BlockService.make[IO](blockAcceptanceManager, addressStorage, blockStorage, transactionStorage, Amount.empty)
+      }
+
+    }
+  }
+
+  test("block should be accepted") { res =>
```

Comment:
`"block should be accepted"` and `"block should not be accepted"` looks weird. maybe:
`valid block should be accepted` and `invalid block should not be accepted`?

Reply:
or even `invalid block should be postponed for acceptance`